Solution1/Solution1.Tests/ConsoleApp/Services/UserCommunicateServiceTests.cs

```diff
@@ -17,81 +24,177 @@ public class UserCommunicateServiceTests
+                    new WeatherForDateDTO() { DateTime = dateStartForecast, Temp = 10, Comment= "It's fresh."},
+                    new WeatherForDateDTO() { DateTime = dateStartForecast.AddDays(1), Temp = 11, Comment= "It's fresh."}
+                }
+            };
+            
+            _invokerMock
+                .Setup(invoker => invoker.RunAsync(It.IsAny<ForecastWeatherCommand>()))
+                .ReturnsAsync(forecastWeather);
 
             //Act
             await _userCommunicationService.CommunicateAsync();
 
-            //Assert
-            var expected = string.Format("Please, enter city name:{0}City name can't be empty.{0}", Environment.NewLine);
+            //Assert            
+            var expected = Menu.GetMenuRepresentation() + Environment.NewLine +
+                "Please, enter city name:" + Environment.NewLine +
+                "Please, enter count day:" + Environment.NewLine +
+                "Minsk weather forecast:" + Environment.NewLine +
+                "Day 0 (January 10, 2022): 10,0 C. It's fresh." + Environment.NewLine +
+                "Day 1 (January 11, 2022): 11,0 C. It's fresh." + Environment.NewLine;
+            
+            _invokerMock.Verify(i => i.RunAsync(It.IsAny<ForecastWeatherCommand>()));
             Assert.Equal(expected, consoleOutput.ToString());
         }
 
         [Fact]
-        public async Task Communicate_EnterCityName_ShowStringRepresentationAsync()
+        public async Task CommunicateAsync_GetCurrentWeather_ShowCurrentWeather()
         {
             // Arrange
             var cityName = "Minsk";
-            var weatherDto = new WeatherDTO() { CityName = cityName, Temp = 10, Comment = "It's fresh." };
-            _weatherServiceMock
-                .Setup(weatherApiService =>
-                    weatherApiService
-                    .GetByCityNameAsync(cityName))
-                .ReturnsAsync(weatherDto);
-
+            var dateStartForecast = new DateTime(2022, 01, 10);
             var consoleOutput = new StringWriter();
             Console.SetOut(consoleOutput);
+            Console.SetIn(new StringReader(string.Format("1{0}{1}{0}", Environment.NewLine, cityName)));
+
+            var Weather = new WeatherDTO()
+            {
+                CityName = cityName,
+                Temp = 10, 
+                Comment = "It's fresh"
+            };
 
-            Console.SetIn(new StringReader(string.Format("Minsk{0}", Environment.NewLine)));
+            _invokerMock
+                .Setup(invoker => invoker.RunAsync(It.IsAny<CurrentWeatherCommand>()))
+                .ReturnsAsync(Weather);
 
             //Act
             await _userCommunicationService.CommunicateAsync();
 
-            //Assert
-            var expected = string.Format("Please, enter city name:{0}In Minsk 10 C. It's fresh.{0}", Environment.NewLine);
+            //Assert            
+            var expected = Menu.GetMenuRepresentation() + Environment.NewLine +
```

Comment:
same - you have different ways of forming strings across the code

Reply:
+